src/Netflix/Netflix.ts

```diff
@@ -0,0 +1,760 @@
+  window.location.href = `https://www.netflix.com/search?q=${encodeURIComponent(
+    title
+  )}`;
+};
+
+// Context
+
+/**
+ * Context is currently disabled becacuse some weird behavior with non-normal
+ * context
+ */
+
+enum NetflixPluginContextEnum {
+  watch = "Netflix Video Player Controls",
+  browse = "Browse Netflix"
+}
+type NetflixPluginContext = NetflixPluginContextEnum | null;
+
+const contextManager = (() => {
+  let enabled = true;
+
+  type ContextPatcher = {
+    add: (context: string) => void;
+    remove: (context: string) => void;
+  };
+
+  /**
+   * Patch context to avoid context duplication
+   * due to addContext being called multiple times causing duplicate
+   * context entry
+   *
+   * Usage:
+   *
+   * ```typescript
+   * patchContext((patcher) => {
+   *  patcher.add("ContextA")
+   *  patcher.add("ContextB")
+   *  patcher.add("ContextC")
+   * });
+   * ```
+   */
+  const patchContext = (fn: (patcher: ContextPatcher) => unknown) => {
+    const contextSet = new Set(PluginBase.util.getContext());
+    const contextPatcher = {
+      add: (context: string) => {
+        if (!contextSet.has(context)) {
+          PluginBase.util.addContext(context);
+        }
+      },
+      remove: (context: string) => {
+        if (contextSet.has(context)) {
+          PluginBase.util.removeContext(context);
+        }
+      }
+    } as ContextPatcher;
+    return fn(contextPatcher);
+  };
+
+  const createContextFromUrl = (url: URL): NetflixPluginContext => {
+    const { pathname } = url;
+    switch (true) {
+      case pathname.startsWith("/watch"):
+        return NetflixPluginContextEnum.watch;
+      case pathname.startsWith("/latest"):
+      case pathname.startsWith("/browse"):
+      case pathname.startsWith("/title"):
+        return NetflixPluginContextEnum.browse;
+      default:
+        return null;
+    }
+  };
+  const setContext = (context: NetflixPluginContext) =>
+    patchContext(patcher => {
+      switch (true) {
+        case context === NetflixPluginContextEnum.browse: {
+          patcher.add(NetflixPluginContextEnum.browse);
+          patcher.add("Normal");
+          return;
+        }
+        case context === NetflixPluginContextEnum.watch: {
+          patcher.add(NetflixPluginContextEnum.watch);
+          patcher.remove("Normal");
+          return;
+        }
+        default: {
+          patcher.remove(NetflixPluginContextEnum.watch);
+          patcher.remove(NetflixPluginContextEnum.browse);
+          patcher.add("Normal");
+          return;
+        }
+      }
+    });
+
+  const refreshCurrentContext = () => {
+    try {
+      setContext(createContextFromUrl(new URL(window.location.href)));
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  return {
+    enable: async () => {
+      enabled = true;
+      while (enabled) {
+        refreshCurrentContext();
+        await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
There's only `popstate` event. No `pushstate` or `locationchange`.
Those events are not fired in this context, so I resorted to this loop

Reply:
> Maybe I need to add this util to the platform.
I'm suggesting to hold until the number of plugins using this pattern reaches 3.
It can be annoying once you have an API that people use and you want it gone.

Reply:
Yes, agree with that.